scripts/compiling.py

```diff
@@ -0,0 +1,322 @@
+import lxml.etree as ET
+import sys
+import os
+
+# shorthand for yaml tabulation
+def getTabs(n):
+	str=""
+	for x in range(n):
+		str+="  "
+	return str
+
+def processSubProperty(prop):
+	if prop.fileName not in msgFiles:
+		subMsg = ""
+		for subProp in prop.properties:
+			if subProp.fileName is not None:
+				processSubProperty(subProp)
+			else:
+				if subProp.unit is not None and subProp.unit == "enum":
+
+					# sort enumeration values for readability
+					for value in sorted( ((v,k) for k,v in subProp.enumeration.iteritems())):
+						subMsg+=subProp.type+" "+value[1]+"="+str(value[0])+"\n"
+			subMsg+=formatProperty(subProp)
+		subFileName = prop.fileName+".msg"
+		msgFiles.append(prop.fileName)
+
+		text_file = open(subFileName, "w")
+		text_file.write(subMsg)
+		text_file.close()
+
+# separate the message generation for messages declared inside services
+def processMessage(module, topic):
+
+	# check if file has already been generated
+	if topic.fileName not in msgFiles:
+
+		global msgFolderPath
+
+		# package folder naming
+		msgFolderPath = os.getcwd()+"/hrim_"+module.type+"_"+module.name+"_msgs/msg"
+
+		# if the package directories don't exist, create them
+		if not os.path.exists(msgFolderPath):
+			os.makedirs(msgFolderPath)
+
+		# position ourselves on the package's msg folder
+		os.chdir(msgFolderPath)
+		msg = ""
+
+		# check for an overall message description
+		if topic.desc is not None and len(topic.desc)>0:
+			msg+="# "+topic.desc+"\n\n"
+
+		for prop in topic.properties:
+
+			if prop.fileName is not None:
+				processSubProperty(prop)
+			else:
+
+				# check for enumeration types
+				if prop.unit is not None and prop.unit == "enum":
+
+					# sort enumeration values for readability
+					for value in sorted( ((v,k) for k,v in prop.enumeration.iteritems())):
+						msg+=prop.type+" "+value[1]+"="+str(value[0])+"\n"
+
+			# process each property, checking if it's value is an array and if it has any description
+			msg+=formatProperty(prop)
+
+		# generate each .msg file and add it to the list
+		if topic.fileName is None:
+			fileName = topic.name.title()+".msg"
+			msgFiles.append(topic.name.title())
+		else:
+			fileName = topic.fileName+".msg"
+			msgFiles.append(topic.fileName)
+
+		text_file = open(fileName, "w")
+		text_file.write(msg)
+		text_file.close()
+
+def formatProperty(prop):
+	if prop.fileName is None:
+		type = dataTypes[prop.type]
+	else:
+		type = msgPkgName+"/"+prop.fileName
+	return type+("[{}] ".format(prop.length if prop.length is not None else "") if prop.array else " ")+prop.name+((" # "+prop.desc) if prop.desc is not None else "")+"\n\n"
+
+def main(module):
+
+	messages = False
+	services = False
+	dependency = False
+
+	# default platform
+	plat = "ros2"
+
+	# map the datatypes
+	global dataTypes
+	dataTypes = {}
+	cwd = os.getcwd()
```

Comment:
if you are not executing `hrim.py` in the same folder probably this is not going to work

Reply:
https://github.com/erlerobot/HRIM/commit/628755067e44aa67c7d7bd352a143c0beb9c6423 should fix this. Makes the script take itself as an "anchor" to initially position itself on it's folder independently of the path from which it was executed.